### PR TITLE
Fix Reader::read

### PR DESCRIPTION
**Pull request by R4PaSs**
With the stream refactor being developed and tested alongside, some buggy behaviours are being discovered.
This PR fixes one of such, where calling read on a dead stream caused the program to segfault.

### CODE REVIEW

**Comment by R4PaSs** [comment]
OK, the failure reports make no sense.
Toaster, do your job, and test this please

File: lib/core/stream.nit [hunk 1]
```diff
@@ -150,6 +150,7 @@ abstract class Reader
 	fun read(i: Int): String do
 		var cs = new CString(i)
```
Comment (privat) [inline lib/core/stream.nit]: assert i >= 0 ?

Comment (R4PaSs) [reply lib/core/stream.nit]: That can be done yep!

**Comment by R4PaSs** [comment]
C'mon Jenkins. test this please

**Pull request merged**